src/ws/cockpitchannelresponse.c

```diff
@@ -76,17 +76,29 @@ cockpit_channel_inject_perform (CockpitChannelInject *inject,
     return;
 
   creds = cockpit_web_service_get_creds (inject->service);
-  application = cockpit_creds_get_application (creds);
+  if (cockpit_web_response_get_url_root (response))
```

Comment:
I thought i made sense to make it available on the response object similar to how path is. Most of the time when something needs to be changed based on a non-standard root, for example when setting a location header, only a response instance is available. We could make it a global in cockpit-ws instead, but since the configured string may need some cleanup to be valid, it felt cleaner to me to do that only once in CockpitWebServer but keep it easily accessible from a response instance.